compiler-rt/lib/xray/xray_interface.cpp

```diff
@@ -57,6 +57,10 @@ static const int16_t cSledLength = 64;
 static const int16_t cSledLength = 8;
 #elif defined(__hexagon__)
 static const int16_t cSledLength = 20;
+#elif SANITIZER_RISCV64
+static const int16_t cSledLength = 76;
+#elif defined(__riscv) && (__riscv_xlen == 32)
```

Comment:
Why the mix? The existing source is odd in mixing the two, but it at least doesn't do so within a single architecture

Reply:
I don't quite follow. What was your suggestion? Using only __riscv_xlen?

Reply:
`SANITIZER_RISCV64` vs `defined(__riscv) && (__riscv_xlen == 32)`

Reply:
We don't have `SANITIZER_RISCV32` I believe, if that's what you meant

Reply:
Rather than adding SANITIZER_RISCV32 shoud we use defined(__riscv) && (__riscv_xlen == 64) instead of SANITIZER_RISCV64?

Reply:
> shoud we use defined(__riscv) && (__riscv_xlen == 64)

I'm fine with that. It's fixed now.